// geb.js/lib/index.js
"use strict";
var __createBinding = (this && this.__createBinding) || (Object.create ? (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    Object.defineProperty(o, k2, { enumerable: true, get: function() { return m[k]; } });
}) : (function(o, m, k, k2) {
    if (k2 === undefined) k2 = k;
    o[k2] = m[k];
}));
var __setModuleDefault = (this && this.__setModuleDefault) || (Object.create ? (function(o, v) {
    Object.defineProperty(o, "default", { enumerable: true, value: v });
}) : function(o, v) {
    o["default"] = v;
});
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (k !== "default" && Object.prototype.hasOwnProperty.call(mod, k)) __createBinding(result, mod, k);
    __setModuleDefault(result, mod);
    return result;
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.BaseContractAPI = exports.MAINNET_ADDRESSES = exports.KOVAN_ADDRESSES = exports.contracts = exports.utils = exports.GebEthersProvider = exports.GebProxyActions = exports.GebErrorTypes = exports.Geb = void 0;
var errors_1 = require("./errors");
Object.defineProperty(exports, "GebErrorTypes", { enumerable: true, get: function () { return errors_1.GebErrorTypes; } });
var proxy_action_1 = require("./proxy-action");
Object.defineProperty(exports, "GebProxyActions", { enumerable: true, get: function () { return proxy_action_1.GebProxyActions; } });
var geb_1 = require("./geb");
Object.defineProperty(exports, "Geb", { enumerable: true, get: function () { return geb_1.Geb; } });
var utils = __importStar(require("./utils"));
exports.utils = utils;
var contracts = __importStar(require("@reflexer-finance/geb-contract-api"));
exports.contracts = contracts;
var geb_ethers_provider_1 = require("@reflexer-finance/geb-ethers-provider");
Object.defineProperty(exports, "GebEthersProvider", { enumerable: true, get: function () { return geb_ethers_provider_1.GebEthersProvider; } });
var geb_contract_base_1 = require("@reflexer-finance/geb-contract-base");
Object.defineProperty(exports, "KOVAN_ADDRESSES", { enumerable: true, get: function () { return geb_contract_base_1.KOVAN_ADDRESSES; } });
Object.defineProperty(exports, "MAINNET_ADDRESSES", { enumerable: true, get: function () { return geb_contract_base_1.MAINNET_ADDRESSES; } });
Object.defineProperty(exports, "BaseContractAPI", { enumerable: true, get: function () { return geb_contract_base_1.BaseContractAPI; } });
