// geb.js/lib/proxy-action-leveraged.js
"use strict";
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GebProxyLeverageActions = void 0;
var geb_contract_base_1 = require("@reflexer-finance/geb-contract-base");
var bignumber_1 = require("@ethersproject/bignumber");
var geb_contract_api_1 = require("@reflexer-finance/geb-contract-api");
var GebProxyLeverageActions = /** @class */ (function (_super) {
    __extends(GebProxyLeverageActions, _super);
    function GebProxyLeverageActions(
    /**
     * Address of the underlying proxy
     */
    proxyAddress, network, chainProvider) {
        var _this = _super.call(this, geb_contract_base_1.getAddressList(network).PROXY_ACTIONS, chainProvider) || this;
        _this.proxyAddress = proxyAddress;
        _this.addressList = geb_contract_base_1.getAddressList(network);
        _this.proxy = new geb_contract_api_1.DsProxy(proxyAddress, _this.chainProvider);
        _this.proxyActionAddress = _this.addressList.PROXY_ACTIONS;
        _this.address = proxyAddress;
        return _this;
    }
    // Override getTransactionRequest to use proxy
    GebProxyLeverageActions.prototype.getTransactionRequest = function (abiFragment, params, ethValue) {
        var data = this.chainProvider.encodeFunctionData(params, abiFragment);
        if (!ethValue) {
            ethValue = bignumber_1.BigNumber.from('0');
        }
        return this.proxy.execute1(ethValue, this.proxyActionAddress, data);
    };
    GebProxyLeverageActions.prototype.allowHandler = function (usr, ok) {
        // prettier-ignore
        return _super.prototype.allowHandler.call(this, this.addressList.SAFE_MANAGER, usr, ok);
    };
    GebProxyLeverageActions.prototype.allowSAFE = function (safe, usr, ok) {
        // prettier-ignore
        return _super.prototype.allowSAFE.call(this, this.addressList.SAFE_MANAGER, safe, usr, ok);
    };
    GebProxyLeverageActions.prototype.approveSAFEModification = function (safeEngine, usr) {
        // prettier-ignore
        return _super.prototype.approveSAFEModification.call(this, safeEngine, usr);
    };
    GebProxyLeverageActions.prototype.coinJoin_join = function (apt, safeHandler, wad) {
        // prettier-ignore
        return _super.prototype.coinJoin_join.call(this, apt, safeHandler, wad);
    };
    GebProxyLeverageActions.prototype.denySAFEModification = function (safeEngine, usr) {
        // prettier-ignore
        return _super.prototype.denySAFEModification.call(this, safeEngine, usr);
    };
    GebProxyLeverageActions.prototype.enterSystem = function (src, safe) {
        // prettier-ignore
        return _super.prototype.enterSystem.call(this, this.addressList.SAFE_MANAGER, src, safe);
    };
    GebProxyLeverageActions.prototype.ethJoin_join1 = function (ethValue, apt, safe) {
        // prettier-ignore
        return _super.prototype.ethJoin_join1.call(this, bignumber_1.BigNumber.from(ethValue), apt, safe);
    };
    GebProxyLeverageActions.prototype.ethJoin_join2 = function (ethValue, apt, safe, value) {
        // prettier-ignore
        return _super.prototype.ethJoin_join2.call(this, bignumber_1.BigNumber.from(ethValue), apt, safe, value);
    };
    GebProxyLeverageActions.prototype.exitETH = function (safe, wad) {
        // prettier-ignore
        return _super.prototype.exitETH.call(this, this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, safe, wad);
    };
    GebProxyLeverageActions.prototype.flashDeleverage = function (uniswapV2Pair, weth, callbackProxy, safe) {
        // prettier-ignore
        return _super.prototype.flashDeleverage.call(this, uniswapV2Pair, this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, this.addressList.GEB_TAX_COLLECTOR, this.addressList.GEB_COIN_JOIN, weth, callbackProxy, safe);
    };
    GebProxyLeverageActions.prototype.flashDeleverageFreeETH = function (uniswapV2Pair, weth, callbackProxy, safe, amountToFree) {
        // prettier-ignore
        return _super.prototype.flashDeleverageFreeETH.call(this, uniswapV2Pair, this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, this.addressList.GEB_TAX_COLLECTOR, this.addressList.GEB_COIN_JOIN, weth, callbackProxy, safe, amountToFree);
    };
    GebProxyLeverageActions.prototype.flashLeverage = function (uniswapV2Pair, weth, callbackProxy, safe, leverage) {
        // prettier-ignore
        return _super.prototype.flashLeverage.call(this, uniswapV2Pair, this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, this.addressList.GEB_TAX_COLLECTOR, this.addressList.GEB_COIN_JOIN, weth, callbackProxy, safe, leverage);
    };
    GebProxyLeverageActions.prototype.freeETH = function (safe, wad) {
        // prettier-ignore
        return _super.prototype.freeETH.call(this, this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, safe, wad);
    };
    GebProxyLeverageActions.prototype.generateDebt = function (safe, wad) {
        // prettier-ignore
        return _super.prototype.generateDebt.call(this, this.addressList.SAFE_MANAGER, this.addressList.GEB_TAX_COLLECTOR, this.addressList.GEB_COIN_JOIN, safe, wad);
    };
    GebProxyLeverageActions.prototype.lockETH = function (ethValue, safe) {
        // prettier-ignore
        return _super.prototype.lockETH.call(this, bignumber_1.BigNumber.from(ethValue), this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, safe);
    };
    GebProxyLeverageActions.prototype.lockETHAndGenerateDebt = function (ethValue, safe, deltaWad) {
        // prettier-ignore
        return _super.prototype.lockETHAndGenerateDebt.call(this, bignumber_1.BigNumber.from(ethValue), this.addressList.SAFE_MANAGER, this.addressList.GEB_TAX_COLLECTOR, this.addressList.GEB_JOIN_ETH_A, this.addressList.GEB_COIN_JOIN, safe, deltaWad);
    };
    GebProxyLeverageActions.prototype.lockETHLeverage = function (ethValue, uniswapV2Pair, weth, callbackProxy, safe, leverage) {
        // prettier-ignore
        return _super.prototype.lockETHLeverage.call(this, bignumber_1.BigNumber.from(ethValue), uniswapV2Pair, this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, this.addressList.GEB_TAX_COLLECTOR, this.addressList.GEB_COIN_JOIN, weth, callbackProxy, safe, leverage);
    };
    GebProxyLeverageActions.prototype.modifySAFECollateralization = function (safe, deltaCollateral, deltaDebt) {
        // prettier-ignore
        return _super.prototype.modifySAFECollateralization.call(this, this.addressList.SAFE_MANAGER, safe, deltaCollateral, deltaDebt);
    };
    GebProxyLeverageActions.prototype.moveSAFE = function (safeSrc, safeDst) {
        // prettier-ignore
        return _super.prototype.moveSAFE.call(this, this.addressList.SAFE_MANAGER, safeSrc, safeDst);
    };
    GebProxyLeverageActions.prototype.openLockETHAndGenerateDebt = function (ethValue, collateralType, deltaWad) {
        // prettier-ignore
        return _super.prototype.openLockETHAndGenerateDebt.call(this, bignumber_1.BigNumber.from(ethValue), this.addressList.SAFE_MANAGER, this.addressList.GEB_TAX_COLLECTOR, this.addressList.GEB_JOIN_ETH_A, this.addressList.GEB_COIN_JOIN, collateralType, deltaWad);
    };
    GebProxyLeverageActions.prototype.openLockETHLeverage = function (ethValue, uniswapV2Pair, weth, callbackProxy, leverage) {
        // prettier-ignore
        return _super.prototype.openLockETHLeverage.call(this, bignumber_1.BigNumber.from(ethValue), uniswapV2Pair, this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, this.addressList.GEB_TAX_COLLECTOR, this.addressList.GEB_COIN_JOIN, weth, callbackProxy, leverage);
    };
    GebProxyLeverageActions.prototype.openSAFE = function (collateralType, usr) {
        // prettier-ignore
        return _super.prototype.openSAFE.call(this, this.addressList.SAFE_MANAGER, collateralType, usr);
    };
    GebProxyLeverageActions.prototype.quitSystem = function (safe, dst) {
        // prettier-ignore
        return _super.prototype.quitSystem.call(this, this.addressList.SAFE_MANAGER, safe, dst);
    };
    GebProxyLeverageActions.prototype.repayDebt = function (safe, wad) {
        // prettier-ignore
        return _super.prototype.repayDebt.call(this, this.addressList.SAFE_MANAGER, this.addressList.GEB_COIN_JOIN, safe, wad);
    };
    GebProxyLeverageActions.prototype.repayDebtAndFreeETH = function (safe, collateralWad, deltaWad) {
        // prettier-ignore
        return _super.prototype.repayDebtAndFreeETH.call(this, this.addressList.SAFE_MANAGER, this.addressList.GEB_JOIN_ETH_A, this.addressList.GEB_COIN_JOIN, safe, collateralWad, deltaWad);
    };
    GebProxyLeverageActions.prototype.transfer = function (collateral, dst, amt) {
        // prettier-ignore
        return _super.prototype.transfer.call(this, collateral, dst, amt);
    };
    GebProxyLeverageActions.prototype.transferCollateral = function (safe, dst, wad) {
        // prettier-ignore
        return _super.prototype.transferCollateral.call(this, this.addressList.SAFE_MANAGER, safe, dst, wad);
    };
    GebProxyLeverageActions.prototype.transferInternalCoins = function (safe, dst, rad) {
        // prettier-ignore
        return _super.prototype.transferInternalCoins.call(this, this.addressList.SAFE_MANAGER, safe, dst, rad);
    };
    GebProxyLeverageActions.prototype.transferSAFEOwnership = function (safe, usr) {
        // prettier-ignore
        return _super.prototype.transferSAFEOwnership.call(this, this.addressList.SAFE_MANAGER, safe, usr);
    };
    GebProxyLeverageActions.prototype.transferSAFEOwnershipToProxy = function (safe, dst) {
        // prettier-ignore
        return _super.prototype.transferSAFEOwnershipToProxy.call(this, this.addressList.PROXY_REGISTRY, this.addressList.SAFE_MANAGER, safe, dst);
    };
    GebProxyLeverageActions.prototype.uniswapV2Call = function (_sender, _amount0, _amount1, _data) {
        // prettier-ignore
        return _super.prototype.uniswapV2Call.call(this, _sender, _amount0, _amount1, _data);
    };
    return GebProxyLeverageActions;
}(geb_contract_api_1.GebProxyLeverageActions));
exports.GebProxyLeverageActions = GebProxyLeverageActions;
