// @reflexer-finance/geb-contract-api/lib/generated/DSValue.js
"use strict";
/* Generated by ts-generator ver. 0.0.8 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.DsValue = void 0;
var geb_contract_base_1 = require("@reflexer-finance/geb-contract-base");
var DsValue = /** @class */ (function (_super) {
    __extends(DsValue, _super);
    function DsValue() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    DsValue.prototype.authority = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "authority", "outputs": [{ "internalType": "contract DSAuthority", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    DsValue.prototype.getResultWithValidity = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "getResultWithValidity", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }, { "internalType": "bool", "name": "", "type": "bool" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    DsValue.prototype.owner = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "owner", "outputs": [{ "internalType": "address", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    DsValue.prototype.read = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "read", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    DsValue.prototype.restartValue = function () {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "restartValue", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, []);
    };
    DsValue.prototype.setAuthority = function (authority_) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "contract DSAuthority", "name": "authority_", "type": "address" }], "name": "setAuthority", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [authority_]);
    };
    DsValue.prototype.setOwner = function (owner_) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "owner_", "type": "address" }], "name": "setOwner", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [owner_]);
    };
    DsValue.prototype.updateResult = function (newMedian) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "newMedian", "type": "uint256" }], "name": "updateResult", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [newMedian]);
    };
    return DsValue;
}(geb_contract_base_1.BaseContractAPI));
exports.DsValue = DsValue;
