// @reflexer-finance/geb-contract-api/lib/generated/EnglishCollateralAuctionHouse.js
"use strict";
/* Generated by ts-generator ver. 0.0.8 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.EnglishCollateralAuctionHouse = void 0;
var geb_contract_base_1 = require("@reflexer-finance/geb-contract-base");
var EnglishCollateralAuctionHouse = /** @class */ (function (_super) {
    __extends(EnglishCollateralAuctionHouse, _super);
    function EnglishCollateralAuctionHouse() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    EnglishCollateralAuctionHouse.prototype.AUCTION_HOUSE_TYPE = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "AUCTION_HOUSE_TYPE", "outputs": [{ "internalType": "bytes32", "name": "", "type": "bytes32" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.AUCTION_TYPE = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "AUCTION_TYPE", "outputs": [{ "internalType": "bytes32", "name": "", "type": "bytes32" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.addAuthorization = function (account) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "account", "type": "address" }], "name": "addAuthorization", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [account]);
    };
    EnglishCollateralAuctionHouse.prototype.amountToRaise = function (id, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "name": "amountToRaise", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [id], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.auctionsStarted = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "auctionsStarted", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.authorizedAccounts = function (address, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "", "type": "address" }], "name": "authorizedAccounts", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [address], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.bidAmount = function (id, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "name": "bidAmount", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [id], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.bidDuration = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "bidDuration", "outputs": [{ "internalType": "uint48", "name": "", "type": "uint48" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.bidIncrease = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "bidIncrease", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.bids = function (uinteger, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "name": "bids", "outputs": [{ "internalType": "uint256", "name": "bidAmount", "type": "uint256" }, { "internalType": "uint256", "name": "amountToSell", "type": "uint256" }, { "internalType": "address", "name": "highBidder", "type": "address" }, { "internalType": "uint48", "name": "bidExpiry", "type": "uint48" }, { "internalType": "uint48", "name": "auctionDeadline", "type": "uint48" }, { "internalType": "address", "name": "forgoneCollateralReceiver", "type": "address" }, { "internalType": "address", "name": "auctionIncomeRecipient", "type": "address" }, { "internalType": "uint256", "name": "amountToRaise", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [uinteger], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.collateralType = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "collateralType", "outputs": [{ "internalType": "bytes32", "name": "", "type": "bytes32" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.decreaseSoldAmount = function (id, amountToBuy, rad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }, { "internalType": "uint256", "name": "amountToBuy", "type": "uint256" }, { "internalType": "uint256", "name": "rad", "type": "uint256" }], "name": "decreaseSoldAmount", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [id, amountToBuy, rad]);
    };
    EnglishCollateralAuctionHouse.prototype.forgoneCollateralReceiver = function (id, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "name": "forgoneCollateralReceiver", "outputs": [{ "internalType": "address", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [id], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.increaseBidSize = function (id, amountToBuy, rad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }, { "internalType": "uint256", "name": "amountToBuy", "type": "uint256" }, { "internalType": "uint256", "name": "rad", "type": "uint256" }], "name": "increaseBidSize", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [id, amountToBuy, rad]);
    };
    EnglishCollateralAuctionHouse.prototype.liquidationEngine = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "liquidationEngine", "outputs": [{ "internalType": "contract LiquidationEngineLike", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.modifyParameters__Bytes32Address = function (parameter, data) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "bytes32", "name": "parameter", "type": "bytes32" }, { "internalType": "address", "name": "data", "type": "address" }], "name": "modifyParameters", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [parameter, data]);
    };
    EnglishCollateralAuctionHouse.prototype.modifyParameters__Bytes32Uint256 = function (parameter, data) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "bytes32", "name": "parameter", "type": "bytes32" }, { "internalType": "uint256", "name": "data", "type": "uint256" }], "name": "modifyParameters", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [parameter, data]);
    };
    EnglishCollateralAuctionHouse.prototype.raisedAmount = function (id, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "name": "raisedAmount", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [id], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.remainingAmountToSell = function (id, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "name": "remainingAmountToSell", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [id], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.removeAuthorization = function (account) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "account", "type": "address" }], "name": "removeAuthorization", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [account]);
    };
    EnglishCollateralAuctionHouse.prototype.restartAuction = function (id) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "name": "restartAuction", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [id]);
    };
    EnglishCollateralAuctionHouse.prototype.safeEngine = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "safeEngine", "outputs": [{ "internalType": "contract SAFEEngineLike", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    EnglishCollateralAuctionHouse.prototype.settleAuction = function (id) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "name": "settleAuction", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [id]);
    };
    EnglishCollateralAuctionHouse.prototype.startAuction = function (forgoneCollateralReceiver, auctionIncomeRecipient, amountToRaise, amountToSell, initialBid) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "forgoneCollateralReceiver", "type": "address" }, { "internalType": "address", "name": "auctionIncomeRecipient", "type": "address" }, { "internalType": "uint256", "name": "amountToRaise", "type": "uint256" }, { "internalType": "uint256", "name": "amountToSell", "type": "uint256" }, { "internalType": "uint256", "name": "initialBid", "type": "uint256" }], "name": "startAuction", "outputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            forgoneCollateralReceiver,
            auctionIncomeRecipient,
            amountToRaise,
            amountToSell,
            initialBid,
        ]);
    };
    EnglishCollateralAuctionHouse.prototype.terminateAuctionPrematurely = function (id) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "id", "type": "uint256" }], "name": "terminateAuctionPrematurely", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [id]);
    };
    EnglishCollateralAuctionHouse.prototype.totalAuctionLength = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "totalAuctionLength", "outputs": [{ "internalType": "uint48", "name": "", "type": "uint48" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    return EnglishCollateralAuctionHouse;
}(geb_contract_base_1.BaseContractAPI));
exports.EnglishCollateralAuctionHouse = EnglishCollateralAuctionHouse;
