// @reflexer-finance/geb-contract-api/lib/generated/GebProxyActions.js
"use strict";
/* Generated by ts-generator ver. 0.0.8 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.GebProxyActions = void 0;
var geb_contract_base_1 = require("@reflexer-finance/geb-contract-base");
var bignumber_1 = require("@ethersproject/bignumber");
var GebProxyActions = /** @class */ (function (_super) {
    __extends(GebProxyActions, _super);
    function GebProxyActions() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    GebProxyActions.prototype.allowHandler = function (manager, usr, ok) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "usr", "type": "address" }, { "internalType": "uint256", "name": "ok", "type": "uint256" }], "name": "allowHandler", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, usr, ok]);
    };
    GebProxyActions.prototype.allowSAFE = function (manager, safe, usr, ok) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "usr", "type": "address" }, { "internalType": "uint256", "name": "ok", "type": "uint256" }], "name": "allowSAFE", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, safe, usr, ok]);
    };
    GebProxyActions.prototype.approveSAFEModification = function (obj, usr) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "obj", "type": "address" }, { "internalType": "address", "name": "usr", "type": "address" }], "name": "approveSAFEModification", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [obj, usr]);
    };
    GebProxyActions.prototype.coinJoin_join = function (apt, safeHandler, wad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "apt", "type": "address" }, { "internalType": "address", "name": "safeHandler", "type": "address" }, { "internalType": "uint256", "name": "wad", "type": "uint256" }], "name": "coinJoin_join", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [apt, safeHandler, wad]);
    };
    GebProxyActions.prototype.denySAFEModification = function (obj, usr) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "obj", "type": "address" }, { "internalType": "address", "name": "usr", "type": "address" }], "name": "denySAFEModification", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [obj, usr]);
    };
    GebProxyActions.prototype.enterSystem = function (manager, src, safe) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "src", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }], "name": "enterSystem", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, src, safe]);
    };
    GebProxyActions.prototype.ethJoin_join = function (ethValue, apt, safe) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "apt", "type": "address" }, { "internalType": "address", "name": "safe", "type": "address" }], "name": "ethJoin_join", "outputs": [], "stateMutability": "payable", "type": "function" };
        return this.getTransactionRequest(abi, [apt, safe], bignumber_1.BigNumber.from(ethValue));
    };
    GebProxyActions.prototype.exitETH = function (manager, ethJoin, safe, wad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "wad", "type": "uint256" }], "name": "exitETH", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, ethJoin, safe, wad]);
    };
    GebProxyActions.prototype.exitTokenCollateral = function (manager, collateralJoin, safe, amt) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "amt", "type": "uint256" }], "name": "exitTokenCollateral", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            collateralJoin,
            safe,
            amt,
        ]);
    };
    GebProxyActions.prototype.freeETH = function (manager, ethJoin, safe, wad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "wad", "type": "uint256" }], "name": "freeETH", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, ethJoin, safe, wad]);
    };
    GebProxyActions.prototype.freeTokenCollateral = function (manager, collateralJoin, safe, amt) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "amt", "type": "uint256" }], "name": "freeTokenCollateral", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            collateralJoin,
            safe,
            amt,
        ]);
    };
    GebProxyActions.prototype.generateDebt = function (manager, taxCollector, coinJoin, safe, wad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "wad", "type": "uint256" }], "name": "generateDebt", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            coinJoin,
            safe,
            wad,
        ]);
    };
    GebProxyActions.prototype.generateDebtAndProtectSAFE = function (manager, taxCollector, coinJoin, safe, wad, liquidationEngine, saviour) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "wad", "type": "uint256" }, { "internalType": "address", "name": "liquidationEngine", "type": "address" }, { "internalType": "address", "name": "saviour", "type": "address" }], "name": "generateDebtAndProtectSAFE", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            coinJoin,
            safe,
            wad,
            liquidationEngine,
            saviour,
        ]);
    };
    GebProxyActions.prototype.lockETH = function (ethValue, manager, ethJoin, safe) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }], "name": "lockETH", "outputs": [], "stateMutability": "payable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, ethJoin, safe], bignumber_1.BigNumber.from(ethValue));
    };
    GebProxyActions.prototype.lockETHAndGenerateDebt = function (ethValue, manager, taxCollector, ethJoin, coinJoin, safe, deltaWad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }], "name": "lockETHAndGenerateDebt", "outputs": [], "stateMutability": "payable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, taxCollector, ethJoin, coinJoin, safe, deltaWad], bignumber_1.BigNumber.from(ethValue));
    };
    GebProxyActions.prototype.lockTokenCollateral = function (manager, collateralJoin, safe, amt, transferFrom) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "amt", "type": "uint256" }, { "internalType": "bool", "name": "transferFrom", "type": "bool" }], "name": "lockTokenCollateral", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            collateralJoin,
            safe,
            amt,
            transferFrom,
        ]);
    };
    GebProxyActions.prototype.lockTokenCollateralAndGenerateDebt = function (manager, taxCollector, collateralJoin, coinJoin, safe, collateralAmount, deltaWad, transferFrom) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "collateralAmount", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }, { "internalType": "bool", "name": "transferFrom", "type": "bool" }], "name": "lockTokenCollateralAndGenerateDebt", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            collateralJoin,
            coinJoin,
            safe,
            collateralAmount,
            deltaWad,
            transferFrom,
        ]);
    };
    GebProxyActions.prototype.lockTokenCollateralGenerateDebtAndProtectSAFE = function (manager, taxCollector, collateralJoin, coinJoin, safe, collateralAmount, deltaWad, transferFrom, liquidationEngine, saviour) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "collateralAmount", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }, { "internalType": "bool", "name": "transferFrom", "type": "bool" }, { "internalType": "address", "name": "liquidationEngine", "type": "address" }, { "internalType": "address", "name": "saviour", "type": "address" }], "name": "lockTokenCollateralGenerateDebtAndProtectSAFE", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            collateralJoin,
            coinJoin,
            safe,
            collateralAmount,
            deltaWad,
            transferFrom,
            liquidationEngine,
            saviour,
        ]);
    };
    GebProxyActions.prototype.makeCollateralBag = function (collateralJoin) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "collateralJoin", "type": "address" }], "name": "makeCollateralBag", "outputs": [{ "internalType": "address", "name": "bag", "type": "address" }], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [collateralJoin]);
    };
    GebProxyActions.prototype.modifySAFECollateralization = function (manager, safe, deltaCollateral, deltaDebt) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "int256", "name": "deltaCollateral", "type": "int256" }, { "internalType": "int256", "name": "deltaDebt", "type": "int256" }], "name": "modifySAFECollateralization", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            safe,
            deltaCollateral,
            deltaDebt,
        ]);
    };
    GebProxyActions.prototype.moveSAFE = function (manager, safeSrc, safeDst) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safeSrc", "type": "uint256" }, { "internalType": "uint256", "name": "safeDst", "type": "uint256" }], "name": "moveSAFE", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, safeSrc, safeDst]);
    };
    GebProxyActions.prototype.openLockETHAndGenerateDebt = function (ethValue, manager, taxCollector, ethJoin, coinJoin, collateralType, deltaWad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "bytes32", "name": "collateralType", "type": "bytes32" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }], "name": "openLockETHAndGenerateDebt", "outputs": [{ "internalType": "uint256", "name": "safe", "type": "uint256" }], "stateMutability": "payable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            ethJoin,
            coinJoin,
            collateralType,
            deltaWad,
        ], bignumber_1.BigNumber.from(ethValue));
    };
    GebProxyActions.prototype.openLockETHGenerateDebtAndProtectSAFE = function (ethValue, manager, taxCollector, ethJoin, coinJoin, collateralType, deltaWad, liquidationEngine, saviour) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "bytes32", "name": "collateralType", "type": "bytes32" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }, { "internalType": "address", "name": "liquidationEngine", "type": "address" }, { "internalType": "address", "name": "saviour", "type": "address" }], "name": "openLockETHGenerateDebtAndProtectSAFE", "outputs": [{ "internalType": "uint256", "name": "safe", "type": "uint256" }], "stateMutability": "payable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            ethJoin,
            coinJoin,
            collateralType,
            deltaWad,
            liquidationEngine,
            saviour,
        ], bignumber_1.BigNumber.from(ethValue));
    };
    GebProxyActions.prototype.openLockGNTAndGenerateDebt = function (manager, taxCollector, gntJoin, coinJoin, collateralType, collateralAmount, deltaWad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "gntJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "bytes32", "name": "collateralType", "type": "bytes32" }, { "internalType": "uint256", "name": "collateralAmount", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }], "name": "openLockGNTAndGenerateDebt", "outputs": [{ "internalType": "address", "name": "bag", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            gntJoin,
            coinJoin,
            collateralType,
            collateralAmount,
            deltaWad,
        ]);
    };
    GebProxyActions.prototype.openLockGNTGenerateDebtAndProtectSAFE = function (manager, taxCollector, gntJoin, coinJoin, collateralType, collateralAmount, deltaWad, liquidationEngine, saviour) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "gntJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "bytes32", "name": "collateralType", "type": "bytes32" }, { "internalType": "uint256", "name": "collateralAmount", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }, { "internalType": "address", "name": "liquidationEngine", "type": "address" }, { "internalType": "address", "name": "saviour", "type": "address" }], "name": "openLockGNTGenerateDebtAndProtectSAFE", "outputs": [{ "internalType": "address", "name": "bag", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            gntJoin,
            coinJoin,
            collateralType,
            collateralAmount,
            deltaWad,
            liquidationEngine,
            saviour,
        ]);
    };
    GebProxyActions.prototype.openLockTokenCollateralAndGenerateDebt = function (manager, taxCollector, collateralJoin, coinJoin, collateralType, collateralAmount, deltaWad, transferFrom) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "bytes32", "name": "collateralType", "type": "bytes32" }, { "internalType": "uint256", "name": "collateralAmount", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }, { "internalType": "bool", "name": "transferFrom", "type": "bool" }], "name": "openLockTokenCollateralAndGenerateDebt", "outputs": [{ "internalType": "uint256", "name": "safe", "type": "uint256" }], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            collateralJoin,
            coinJoin,
            collateralType,
            collateralAmount,
            deltaWad,
            transferFrom,
        ]);
    };
    GebProxyActions.prototype.openLockTokenCollateralGenerateDebtAndProtectSAFE = function (manager, taxCollector, collateralJoin, coinJoin, collateralType, collateralAmount, deltaWad, transferFrom, liquidationEngine, saviour) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "taxCollector", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "bytes32", "name": "collateralType", "type": "bytes32" }, { "internalType": "uint256", "name": "collateralAmount", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }, { "internalType": "bool", "name": "transferFrom", "type": "bool" }, { "internalType": "address", "name": "liquidationEngine", "type": "address" }, { "internalType": "address", "name": "saviour", "type": "address" }], "name": "openLockTokenCollateralGenerateDebtAndProtectSAFE", "outputs": [{ "internalType": "uint256", "name": "safe", "type": "uint256" }], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            taxCollector,
            collateralJoin,
            coinJoin,
            collateralType,
            collateralAmount,
            deltaWad,
            transferFrom,
            liquidationEngine,
            saviour,
        ]);
    };
    GebProxyActions.prototype.openSAFE = function (manager, collateralType, usr) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "bytes32", "name": "collateralType", "type": "bytes32" }, { "internalType": "address", "name": "usr", "type": "address" }], "name": "openSAFE", "outputs": [{ "internalType": "uint256", "name": "safe", "type": "uint256" }], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, collateralType, usr]);
    };
    GebProxyActions.prototype.protectSAFE = function (manager, safe, liquidationEngine, saviour) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "liquidationEngine", "type": "address" }, { "internalType": "address", "name": "saviour", "type": "address" }], "name": "protectSAFE", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            safe,
            liquidationEngine,
            saviour,
        ]);
    };
    GebProxyActions.prototype.quitSystem = function (manager, safe, dst) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "dst", "type": "address" }], "name": "quitSystem", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, safe, dst]);
    };
    GebProxyActions.prototype.repayAllDebt = function (manager, coinJoin, safe) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }], "name": "repayAllDebt", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, coinJoin, safe]);
    };
    GebProxyActions.prototype.repayAllDebtAndFreeETH = function (manager, ethJoin, coinJoin, safe, collateralWad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "collateralWad", "type": "uint256" }], "name": "repayAllDebtAndFreeETH", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            ethJoin,
            coinJoin,
            safe,
            collateralWad,
        ]);
    };
    GebProxyActions.prototype.repayAllDebtAndFreeTokenCollateral = function (manager, collateralJoin, coinJoin, safe, collateralAmount) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "collateralAmount", "type": "uint256" }], "name": "repayAllDebtAndFreeTokenCollateral", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            collateralJoin,
            coinJoin,
            safe,
            collateralAmount,
        ]);
    };
    GebProxyActions.prototype.repayDebt = function (manager, coinJoin, safe, wad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "wad", "type": "uint256" }], "name": "repayDebt", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, coinJoin, safe, wad]);
    };
    GebProxyActions.prototype.repayDebtAndFreeETH = function (manager, ethJoin, coinJoin, safe, collateralWad, deltaWad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "collateralWad", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }], "name": "repayDebtAndFreeETH", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            ethJoin,
            coinJoin,
            safe,
            collateralWad,
            deltaWad,
        ]);
    };
    GebProxyActions.prototype.repayDebtAndFreeTokenCollateral = function (manager, collateralJoin, coinJoin, safe, collateralAmount, deltaWad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "collateralAmount", "type": "uint256" }, { "internalType": "uint256", "name": "deltaWad", "type": "uint256" }], "name": "repayDebtAndFreeTokenCollateral", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            collateralJoin,
            coinJoin,
            safe,
            collateralAmount,
            deltaWad,
        ]);
    };
    GebProxyActions.prototype.safeLockETH = function (ethValue, manager, ethJoin, safe, owner) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "ethJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "owner", "type": "address" }], "name": "safeLockETH", "outputs": [], "stateMutability": "payable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, ethJoin, safe, owner], bignumber_1.BigNumber.from(ethValue));
    };
    GebProxyActions.prototype.safeLockTokenCollateral = function (manager, collateralJoin, safe, amt, transferFrom, owner) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "collateralJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "amt", "type": "uint256" }, { "internalType": "bool", "name": "transferFrom", "type": "bool" }, { "internalType": "address", "name": "owner", "type": "address" }], "name": "safeLockTokenCollateral", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            collateralJoin,
            safe,
            amt,
            transferFrom,
            owner,
        ]);
    };
    GebProxyActions.prototype.safeRepayAllDebt = function (manager, coinJoin, safe, owner) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "owner", "type": "address" }], "name": "safeRepayAllDebt", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, coinJoin, safe, owner]);
    };
    GebProxyActions.prototype.safeRepayDebt = function (manager, coinJoin, safe, wad, owner) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "address", "name": "coinJoin", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "uint256", "name": "wad", "type": "uint256" }, { "internalType": "address", "name": "owner", "type": "address" }], "name": "safeRepayDebt", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            manager,
            coinJoin,
            safe,
            wad,
            owner,
        ]);
    };
    GebProxyActions.prototype.tokenCollateralJoin_join = function (apt, safe, amt, transferFrom) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "apt", "type": "address" }, { "internalType": "address", "name": "safe", "type": "address" }, { "internalType": "uint256", "name": "amt", "type": "uint256" }, { "internalType": "bool", "name": "transferFrom", "type": "bool" }], "name": "tokenCollateralJoin_join", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [apt, safe, amt, transferFrom]);
    };
    GebProxyActions.prototype.transfer = function (collateral, dst, amt) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "collateral", "type": "address" }, { "internalType": "address", "name": "dst", "type": "address" }, { "internalType": "uint256", "name": "amt", "type": "uint256" }], "name": "transfer", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [collateral, dst, amt]);
    };
    GebProxyActions.prototype.transferCollateral = function (manager, safe, dst, wad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "dst", "type": "address" }, { "internalType": "uint256", "name": "wad", "type": "uint256" }], "name": "transferCollateral", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, safe, dst, wad]);
    };
    GebProxyActions.prototype.transferInternalCoins = function (manager, safe, dst, rad) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "dst", "type": "address" }, { "internalType": "uint256", "name": "rad", "type": "uint256" }], "name": "transferInternalCoins", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, safe, dst, rad]);
    };
    GebProxyActions.prototype.transferSAFEOwnership = function (manager, safe, usr) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "usr", "type": "address" }], "name": "transferSAFEOwnership", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [manager, safe, usr]);
    };
    GebProxyActions.prototype.transferSAFEOwnershipToProxy = function (proxyRegistry, manager, safe, dst) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "proxyRegistry", "type": "address" }, { "internalType": "address", "name": "manager", "type": "address" }, { "internalType": "uint256", "name": "safe", "type": "uint256" }, { "internalType": "address", "name": "dst", "type": "address" }], "name": "transferSAFEOwnershipToProxy", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [
            proxyRegistry,
            manager,
            safe,
            dst,
        ]);
    };
    return GebProxyActions;
}(geb_contract_base_1.BaseContractAPI));
exports.GebProxyActions = GebProxyActions;
