// @reflexer-finance/geb-contract-api/lib/generated/RateSetter.js
"use strict";
/* Generated by ts-generator ver. 0.0.8 */
/* tslint:disable */
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (Object.prototype.hasOwnProperty.call(b, p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
Object.defineProperty(exports, "__esModule", { value: true });
exports.RateSetter = void 0;
var geb_contract_base_1 = require("@reflexer-finance/geb-contract-base");
var RateSetter = /** @class */ (function (_super) {
    __extends(RateSetter, _super);
    function RateSetter() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    RateSetter.prototype.RAY = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "RAY", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.WAD = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "WAD", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.addAuthorization = function (account) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "account", "type": "address" }], "name": "addAuthorization", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [account]);
    };
    RateSetter.prototype.addition__Uint256Uint256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "addition", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.addition__Int256Int256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "int256", "name": "x", "type": "int256" }, { "internalType": "int256", "name": "y", "type": "int256" }], "name": "addition", "outputs": [{ "internalType": "int256", "name": "z", "type": "int256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.addition__Uint256Int256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "int256", "name": "y", "type": "int256" }], "name": "addition", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.authorizedAccounts = function (address, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "", "type": "address" }], "name": "authorizedAccounts", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [address], multicall);
    };
    RateSetter.prototype.baseUpdateCallerReward = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "baseUpdateCallerReward", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.contractEnabled = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "contractEnabled", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.disableContract = function () {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "disableContract", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, []);
    };
    RateSetter.prototype.getCallerReward = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "getCallerReward", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.getRedemptionAndMarketPrices = function () {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "getRedemptionAndMarketPrices", "outputs": [{ "internalType": "uint256", "name": "marketPrice", "type": "uint256" }, { "internalType": "uint256", "name": "redemptionPrice", "type": "uint256" }], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, []);
    };
    RateSetter.prototype.lastUpdateTime = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "lastUpdateTime", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.latestMarketPrice = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "latestMarketPrice", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.maxUpdateCallerReward = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "maxUpdateCallerReward", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.minimum = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "minimum", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.modifyParameters__Bytes32Address = function (parameter, addr) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "bytes32", "name": "parameter", "type": "bytes32" }, { "internalType": "address", "name": "addr", "type": "address" }], "name": "modifyParameters", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [parameter, addr]);
    };
    RateSetter.prototype.modifyParameters__Bytes32Uint256 = function (parameter, val) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "bytes32", "name": "parameter", "type": "bytes32" }, { "internalType": "uint256", "name": "val", "type": "uint256" }], "name": "modifyParameters", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [parameter, val]);
    };
    RateSetter.prototype.multiply__Uint256Uint256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "multiply", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.multiply__Int256Int256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "int256", "name": "x", "type": "int256" }, { "internalType": "int256", "name": "y", "type": "int256" }], "name": "multiply", "outputs": [{ "internalType": "int256", "name": "z", "type": "int256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.multiply__Int256Uint256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "int256", "name": "x", "type": "int256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "multiply", "outputs": [{ "internalType": "int256", "name": "z", "type": "int256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.oracleRelayer = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "oracleRelayer", "outputs": [{ "internalType": "contract OracleRelayerLike", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.orcl = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "orcl", "outputs": [{ "internalType": "contract OracleLike", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.perSecondCallerRewardIncrease = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "perSecondCallerRewardIncrease", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.pidCalculator = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "pidCalculator", "outputs": [{ "internalType": "contract PIDCalculator", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.rad = function (x, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }], "name": "rad", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x], multicall);
    };
    RateSetter.prototype.ray = function (x, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }], "name": "ray", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x], multicall);
    };
    RateSetter.prototype.rdivide = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "rdivide", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.removeAuthorization = function (account) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "account", "type": "address" }], "name": "removeAuthorization", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [account]);
    };
    RateSetter.prototype.rmultiply__Uint256Uint256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "rmultiply", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.rmultiply__Uint256Int256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "int256", "name": "y", "type": "int256" }], "name": "rmultiply", "outputs": [{ "internalType": "int256", "name": "z", "type": "int256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.rpower = function (x, n, base, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "n", "type": "uint256" }, { "internalType": "uint256", "name": "base", "type": "uint256" }], "name": "rpower", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, n, base], multicall);
    };
    RateSetter.prototype.subtract__Uint256Uint256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "subtract", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.subtract__Int256Int256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "int256", "name": "x", "type": "int256" }, { "internalType": "int256", "name": "y", "type": "int256" }], "name": "subtract", "outputs": [{ "internalType": "int256", "name": "z", "type": "int256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.treasury = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "treasury", "outputs": [{ "internalType": "contract StabilityFeeTreasuryLike", "name": "", "type": "address" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.treasuryAllowance = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "treasuryAllowance", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.updateRate = function (feeReceiver) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "address", "name": "feeReceiver", "type": "address" }], "name": "updateRate", "outputs": [], "stateMutability": "nonpayable", "type": "function" };
        return this.getTransactionRequest(abi, [feeReceiver]);
    };
    RateSetter.prototype.updateRateDelay = function (multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [], "name": "updateRateDelay", "outputs": [{ "internalType": "uint256", "name": "", "type": "uint256" }], "stateMutability": "view", "type": "function" };
        return this.ethCallOrMulticall(abi, [], multicall);
    };
    RateSetter.prototype.wdivide = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "wdivide", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.wmultiply__Uint256Uint256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "uint256", "name": "x", "type": "uint256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "wmultiply", "outputs": [{ "internalType": "uint256", "name": "z", "type": "uint256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    RateSetter.prototype.wmultiply__Int256Uint256 = function (x, y, multicall) {
        // prettier-ignore
        // @ts-ignore
        var abi = { "inputs": [{ "internalType": "int256", "name": "x", "type": "int256" }, { "internalType": "uint256", "name": "y", "type": "uint256" }], "name": "wmultiply", "outputs": [{ "internalType": "int256", "name": "z", "type": "int256" }], "stateMutability": "pure", "type": "function" };
        return this.ethCallOrMulticall(abi, [x, y], multicall);
    };
    return RateSetter;
}(geb_contract_base_1.BaseContractAPI));
exports.RateSetter = RateSetter;
